### ISSUE TITLE
Improvement/Fix: ShallowReference / HibernateUnproxyObject 

### ISSUE THREAD

**Issue by f-aubert** [javers/javers#558]
I noticed another small thing, and although not a big problem, I'd like to reference it. When dealing with shallow references, it should be possible, at least in theory, to work with hibernate proxy and not having to hit the database just to compare the id itself as per shallow reference definition... As of now, 3.3.2, I notice each commit involving a change in an Entity referring another one with ShallowReference on a Proxy triggers the load of the proxy... Thx for your time.

**Comment by bartoszwalacik** [comment]
ok, could you push a test case with a reproduction of this issue?

**Comment by f-aubert** [comment]
Here is my first attempt at uploading a Patch to apply to Javers in order to demonstrate my UseCase. Please forgive me if I did miss something. I can correct it and submit it again, as now I have a working Javers checkout. It works best to run tests one at a time.
[Issue558_Test_Case.patch.zip](https://github.com/javers/javers/files/1115699/Issue558_Test_Case.patch.zip)

**Comment by bartoszwalacik** [comment]
It would be easier if you push this code to github (to your fork of javers repository)

**Comment by f-aubert** [comment]
I didn't intend to push one more project "and pollute github" just to show my test case, but if it helps, I will do it tomorrow.

**Comment by bartoszwalacik** [comment]
Fork is not another project, it's more like a branch

**Comment by f-aubert** [comment]
Although I have the feeling to be copying your work, just for the sake of showing you a test case, I did it. https://github.com/f-aubert/javers

**Comment by bartoszwalacik** [comment]
That's fine, thanks , that's the way we work on github

**Comment by bartoszwalacik** [comment]
just a link to your [commit](https://github.com/f-aubert/javers/commit/f12cc8b69f5540742ae775c03f563e8456231ad3)

**Comment by bartoszwalacik** [comment]
Ok, so looks like we can get the id and persistent class from a proxy without initializing it. So the issue is doable

        println "proxy.isInitialized: " + Hibernate.isInitialized(mathsRef)
        println "proxy.class: " + mathsRef.getClass()
        println "proxy.id: " + mathsRef.getId()
        println "proxy.persistenClass: "+ mathsRef.getHibernateLazyInitializer().getPersistentClass()
        println "proxy.isInitialized: " + Hibernate.isInitialized(mathsRef)
        println 'I am happy :)'

```
proxy.isInitialized: false
proxy.class: class org.javers.spring.boot.sql.Course_$$_jvstfba_1
proxy.id: 1
proxy.persistenClass: class org.javers.spring.boot.sql.Course
proxy.isInitialized: false
I am happy :)
```

**Comment by bartoszwalacik** [comment]
fix released in 3.7.9
